source/ngf-mtl/impl.mm

```diff
@@ -1831,12 +1891,19 @@ ngf_error ngf_cmd_begin_render_pass(
             clear_info->clear_color[2],
             clear_info->clear_color[3]);
       }
-      mtl_desc.resolveTexture =
-          rt->is_default ? CURRENT_CONTEXT->frame.resolve_attachment_texture() : nil;
-      if (mtl_desc.resolveTexture) {
-        // Override user-specified store action
-        mtl_desc.storeAction = MTLStoreActionMultisampleResolve;
+
+      if (attachment_desc.sample_count > NGF_SAMPLE_COUNT_1) {
+        if (rt->is_default) {
+          mtl_desc.resolveTexture = CURRENT_CONTEXT->frame.resolve_attachment_texture();
+          // Override user-specified store action
```

Comment:
thinking about this a little more, we should stop overriding the user-specified store action completely. this was a holdover from when MSAA was only supported for the default render target. now that the resolve store action is exposed in the API, users should be setting it appropriately when initiating the render pass.